src/librustc_mir/interpret/operand.rs

```diff
@@ -335,6 +335,17 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         }
     }
 
+    /// Read vector length and element type
+    pub fn read_vector_ty(
```

Comment:
This doesn't even look at the actual "operand", right? It is just a layout function?

Then it (a) should be mixed up with all the other `read_*` methods as it is very different, and (b) shouldn't even be called "read" as it doesn't read from memory.

I am not sure what the best place is for such a method, but it is not `operand.rs`. Sounds more like a `TyLayout` method to me?